substra/sdk/client.py

```diff
@@ -78,19 +217,74 @@ class Client:
 
     def __init__(
         self,
+        *,
+        configuration_name: Optional[str] = None,
+        configuration_file: Optional[pathlib.Path] = None,
         url: Optional[str] = None,
         token: Optional[str] = None,
-        retry_timeout: int = DEFAULT_RETRY_TIMEOUT,
-        insecure: bool = False,
-        backend_type: schemas.BackendType = schemas.BackendType.REMOTE,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        retry_timeout: Optional[int] = None,
+        insecure: Optional[bool] = None,
+        backend_type: Optional[schemas.BackendType] = None,
     ):
-        self._retry_timeout = retry_timeout
-        self._token = token
+        if configuration_file and not configuration_name:
+            logger.info("Configuration file ignored because no `configuration_name` was given.")
+
+        code_values = {
+            "url": url,
+            "token": token,
+            "username": username,
+            "password": password,
+            "retry_timeout": retry_timeout,
+            "insecure": insecure,
+            "backend_type": backend_type,
+        }
+        config_dict = get_client_configuration(
+            configuration_name=configuration_name, config_file=configuration_file, code_values=code_values
+        )
+        if config_dict["backend_type"].value is None:
+            logger.info("No backend type specified, defaulting to subprocess")
+            backend_type = schemas.BackendType.LOCAL_SUBPROCESS
```

Comment:
What do you think about storing `schemas.BackendType.LOCAL_SUBPROCESS` on top of the file in a `DEFAULT_BACKEND_TYPE` to match the logic of `DEFAULT_RETRY_TIMEOUT` for instance ?